qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -367,3 +376,270 @@ def pass_manager(self, pass_manager_config, optimization_level=None) -> PassMana
+                max_trials=2500,  # Limits layout scoring to < 600ms on ~400 qubit devices
+            )
+            layout.append(choose_layout_1, condition=_layout_not_perfect)
+            choose_layout_2 = SabreLayout(
+                coupling_map,
+                max_iterations=2,
+                seed=pass_manager_config.seed_transpiler,
+                swap_trials=5,
+                layout_trials=5,
+                skip_routing=pass_manager_config.routing_method is not None
+                and pass_manager_config.routing_method != "sabre",
+            )
+            layout.append(
+                [BarrierBeforeFinalMeasurements(), choose_layout_2], condition=_vf2_match_not_found
+            )
+        elif optimization_level == 2:
+            choose_layout_0 = VF2Layout(
+                coupling_map=pass_manager_config.coupling_map,
+                seed=pass_manager_config.seed_transpiler,
+                call_limit=int(5e6),  # Set call limit to ~10s with rustworkx 0.10.2
+                properties=pass_manager_config.backend_properties,
+                target=pass_manager_config.target,
+                max_trials=25000,  # Limits layout scoring to < 10s on ~400 qubit devices
+            )
+            layout.append(choose_layout_0, condition=_choose_layout_condition)
+            choose_layout_1 = SabreLayout(
+                coupling_map,
+                max_iterations=2,
+                seed=pass_manager_config.seed_transpiler,
+                swap_trials=10,
+                layout_trials=10,
+                skip_routing=pass_manager_config.routing_method is not None
+                and pass_manager_config.routing_method != "sabre",
+            )
+            layout.append(
+                [BarrierBeforeFinalMeasurements(), choose_layout_1], condition=_vf2_match_not_found
+            )
+        elif optimization_level == 3:
+            choose_layout_0 = VF2Layout(
+                coupling_map=pass_manager_config.coupling_map,
+                seed=pass_manager_config.seed_transpiler,
+                call_limit=int(3e7),  # Set call limit to ~60s with rustworkx 0.10.2
+                properties=pass_manager_config.backend_properties,
+                target=pass_manager_config.target,
+                max_trials=250000,  # Limits layout scoring to < 60s on ~400 qubit devices
+            )
+            layout.append(choose_layout_0, condition=_choose_layout_condition)
+            choose_layout_1 = SabreLayout(
+                coupling_map,
+                max_iterations=4,
+                seed=pass_manager_config.seed_transpiler,
+                swap_trials=20,
+                layout_trials=20,
+                skip_routing=pass_manager_config.routing_method is not None
+                and pass_manager_config.routing_method != "sabre",
+            )
+            layout.append(
+                [BarrierBeforeFinalMeasurements(), choose_layout_1], condition=_vf2_match_not_found
+            )
+
```

Comment:
We're making an implicit choice here to return _nothing_ if `optimization_level` isn't set, or if it's set to an invalid value.  Is that something intentional? - it feels wrong.

Reply:
The plugin interface guarantees it's only ever 0, 1, 2, or 3. But I can add an explicit exception for `else`.

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/10622/commits/8bb74eda0b4322cba0baa89c082c2bafbe187dda